opentelemetry-sdk/src/opentelemetry/sdk/context/propagation/b3_format.py

```diff
@@ -111,17 +111,18 @@ def inject(cls, context, set_in_carrier, carrier):
         set_in_carrier(carrier, cls.SAMPLED_KEY, "1" if sampled else "0")
 
 
-def format_trace_id(trace_id: int):
+def format_trace_id(trace_id: int) -> str:
     """Format the trace id according to b3 specification."""
     return format(trace_id, "032x")
 
 
-def format_span_id(span_id: int):
+def format_span_id(span_id: int) -> str:
     """Format the span id according to b3 specification."""
     return format(span_id, "016x")
 
 
-def _extract_first_element(list_object: list) -> typing.Optional[object]:
-    if list_object:
-        return list_object[0]
-    return None
+_T = typing.TypeVar("_T")
+
+
+def _extract_first_element(items: typing.Iterable[_T]) -> typing.Optional[_T]:
+    return next(iter(items), None)
```

Comment:
This changes the behavior of the function, it'll throw now if `items` is null. The type declaration doesn't help since it's called above (in `extract`) with an unannotated arg.